scalyr_agent/log_processing.py

```diff
@@ -1848,14 +1857,22 @@ def process_line(self, input_line):
 
         return input_line, modified_it
 
-    def add_redaction_rule(self, redaction_expression, replacement_text):
+    def add_redaction_rule(self, redaction_expression, replacement_text, hash_salt=''):
         """Appends a new redaction rule to this instance.
 
         @param redaction_expression: The regular expression that must match some portion of the line.
         @param replacement_text: The text to replace the matched text with. May include \1 etc to use a portion of the
             matched text.
+        @param hash_salt: If hashing is set, then the cryptographic salt to be used
+
+        @type hash_salt: str (optional)
```

Comment:
Don't use the (optional) string here.  This will (I believe) get in the way of PyCharm's ability to use `str` as the expected type of the parameter.